File: clients/Go/model/primitive_type.go
package model

import "io"
import . "aicup2020/stream"

type PrimitiveType int32
const (
    PrimitiveTypeLines PrimitiveType = 0
    PrimitiveTypeTriangles PrimitiveType = 1
)
func ReadPrimitiveType(reader io.Reader) PrimitiveType {
    switch ReadInt32(reader) {
    case 0:
        return PrimitiveTypeLines
    case 1:
        return PrimitiveTypeTriangles
    }
    panic("Unexpected tag value")
}

File: clients/Scala/src/main/scala/model/PrimitiveType.scala
package model

import util.StreamUtil

sealed abstract class PrimitiveType (val tag: Int) {
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, tag)
    }
}

object PrimitiveType {
    case object LINES extends PrimitiveType(0)
    case object TRIANGLES extends PrimitiveType(1)
    def readFrom(stream: java.io.InputStream): PrimitiveType = StreamUtil.readInt(stream) match {
        case 0 => LINES
        case 1 => TRIANGLES
        case _ => throw new java.io.IOException("Unexpected tag value")
    }
}
